cpp/src/lance/io/filter_test.cc

```diff
@@ -0,0 +1,90 @@
+//  Copyright 2022 Lance Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+#include "lance/io/filter.h"
+
+#include <arrow/array.h>
+#include <arrow/compute/exec/expression.h>
+#include <arrow/record_batch.h>
+#include <fmt/format.h>
+
+#include <catch2/catch_test_macros.hpp>
+
+#include "lance/arrow/stl.h"
+#include "lance/arrow/type.h"
+#include "lance/format/schema.h"
+
+using ::arrow::compute::equal;
+using ::arrow::compute::field_ref;
+using ::arrow::compute::literal;
+using ::arrow::compute::or_;
+
+const auto kSchema =
+    lance::format::Schema(::arrow::schema({::arrow::field("pk", ::arrow::int32()),
+                                           ::arrow::field("value", ::arrow::int32()),
+                                           ::arrow::field("label", ::arrow::utf8())}));
+
+TEST_CASE("Test without filter") {
+  auto empty_filter =
+      lance::io::Filter::Make(kSchema, ::arrow::compute::literal(true)).ValueOrDie();
+  CHECK(empty_filter == nullptr);
+}
+
+TEST_CASE("Test with one condition") {
+  auto filter = lance::io::Filter::Make(kSchema, equal(field_ref("value"), literal("32")));
```

Comment:
Ok this just checks the filter construction and doesn't actually apply the filter right

Reply:
yes, this is just creating filters